app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/AccountSetupAccountType.kt

```diff
@@ -42,47 +48,60 @@ class AccountSetupAccountType : K9Activity() {
         val accountUuid = intent.getStringExtra(EXTRA_ACCOUNT) ?: error("No account UUID provided")
         account = preferences.getAccount(accountUuid) ?: error("No account with given UUID found")
         makeDefault = intent.getBooleanExtra(EXTRA_MAKE_DEFAULT, false)
+        defaultAccountSettings = intent.getSerializableExtra(EXTRA_DEFAULT_SETTINGS) as DefaultAccountSettings
     }
 
     private fun setupPop3Account() {
-        setupAccount(Protocols.POP3, "pop3+ssl+")
+        setupAccount(Protocols.POP3)
     }
 
     private fun setupImapAccount() {
-        setupAccount(Protocols.IMAP, "imap+ssl+")
+        setupAccount(Protocols.IMAP)
     }
 
-    private fun setupAccount(serverType: String, schemePrefix: String) {
-        setupStoreAndSmtpTransport(serverType, schemePrefix)
+    private fun setupAccount(serverType: String) {
+        setupStoreAndSmtpTransport(serverType)
         createSpecialLocalFolders()
         returnAccountTypeSelectionResult()
     }
 
-    private fun setupStoreAndSmtpTransport(serverType: String, schemePrefix: String) {
+    private fun setupStoreAndSmtpTransport(serverType: String) {
```

Comment:
With the introduction of `DefaultAccountSettings` we can now move the store/transport URI creation to `AccountSetupIncoming` and `AccountSetupOutgoing`. We just need to pass the `DefaultAccountSettings` to those activities.

Reply:
The store URI creation is done in `SetupAccountType` instead of `SetupIncoming` because there is no way to know inside `SetupIncoming` if it should be restored to the initial settings. One example is visiting the `SetupIncoming` screen to edit the settings (as opposed to during initial setup) as this screen is used in that scenario also, another is after going back to `SetupAccountType` then into `SetupIncoming` for a second time during setup.
The best method I've found for determining when the initial settings should be restore is when your on the previous step. I.e. if your on `SetupAccountType` it means you can delete any previous incoming settings and restore the initial settings.

The reason the transport URI is also setup in `SetupAccountType` and not differed until `SetupIncoming` is that the account transport URI settings are required in `SetupIncoming` during the connection testing step.

Reply:
> The store URI creation is done in `SetupAccountType` instead of `SetupIncoming` because there is no way to know inside `SetupIncoming` if it should be restored to the initial settings. One example is visiting the `SetupIncoming` screen to edit the settings (as opposed to during initial setup) as this screen is used in that scenario also, another is after going back to `SetupAccountType` then into `SetupIncoming` for a second time during setup.

The Intent action is different when `AccountSetupIncoming` is started during setup vs. when it is started to edit the current settings. 
When going back in the setup workflow we don't launch a new copy of `AccountSetupIncoming`, we return to the old one. That means the activity is rarely restarted and we don't even hit the code to set initial values. But if the activity is restarted `savedInstanceState` won't be `null`.
Together this information could be used to know when to set the initial account values. But I just had a closer look at `AccountSetupIncoming` and realized it would need a lot more work. Since we want to rewrite the setup code anyway, it's not worth doing much extra work now. So we'll stick with your current approach :+1: